runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineResult.java

```diff
@@ -104,16 +104,15 @@ protected abstract State awaitTermination(Duration duration)
   @Override
   public State waitUntilFinish(final Duration duration) {
     try {
-      state = awaitTermination(duration);
+      State finishState = awaitTermination(duration);
```

Comment:
IMO its just weird, I kept looking for what you were doing with 'finishState' but couldn't find it, but that's really up to you.